src/wp-includes/blocks/gallery.php

```diff
@@ -77,18 +81,9 @@ function block_core_gallery_render( $attributes, $content ) {
 	}
 
 	// Set the CSS variable to the column value, and the `gap` property to the combined gap value.
-	$style = '.' . $class . '{ --wp--style--unstable-gallery-gap: ' . $gap_column . '; gap: ' . $gap_value . '}';
+	$style = '.wp-block-gallery.' . $class . '{ --wp--style--unstable-gallery-gap: ' . $gap_column . '; gap: ' . $gap_value . '}';
 
-	// Ideally styles should be loaded in the head, but blocks may be parsed
-	// after that, so loading in the footer for now.
-	// See https://core.trac.wordpress.org/ticket/53494.
-	add_action(
-		'wp_footer',
-		function () use ( $style ) {
-			echo '<style> ' . $style . '</style>';
-		},
-		11
-	);
+	gutenberg_enqueue_block_support_styles( $style, 11 );
```

Comment:
`gutenberg_` prefix ⚡

Reply:
great catch, we need to patch it on the Gutenberg side 👍🏻

Reply:
`gutenberg_enqueue_block_support_styles` actually in [`lib/compat-wordpress-6.1`](https://github.com/WordPress/gutenberg/blob/132b86cd48e4f785a249175476302c66dc95ed2c/lib/compat/wordpress-6.1/script-loader.php#L8-L37):

> This method was added to core in 5.9.1, but with a single param ($style). The second param ($priority) was
> added post 6.0, so the 6.1 release needs to have wp_enqueue_block_support_styles updated to include this param.

Reply:
Backport PR: https://github.com/WordPress/wordpress-develop/pull/3158

Reply:
> Backport PR: https://github.com/WordPress/wordpress-develop/pull/3158

GB counterpart: https://github.com/WordPress/gutenberg/pull/43779

Both have been merged. We now need to wait for the next round of `@wordpress/` packages to be published from GB to pick up the changes from https://github.com/WordPress/gutenberg/pull/43779.

Reply:
I left my comment on the merged PR. We should do as much as possible to avoid leaking any Gutenberg-specific code into WordPress core.

Reply:
> I left my comment on the merged PR. We should do as much as possible to avoid leaking any Gutenberg-specific code into WordPress core.

Thanks Grzegorz! I reverted https://github.com/WordPress/gutenberg/pull/43779 in https://github.com/WordPress/gutenberg/pull/43857 and filed https://github.com/WordPress/gutenberg/pull/43859 as an alternative approach.